feature-libs/checkout/base/components/components/delivery-mode/delivery-mode.component.spec.ts

```diff
@@ -0,0 +1,243 @@
+        ],
+      }).compileComponents();
+
+      checkoutDeliveryModesFacade = TestBed.inject(CheckoutDeliveryModesFacade);
+      checkoutConfigService = TestBed.inject(CheckoutConfigService);
+      checkoutStepService = TestBed.inject(
+        CheckoutStepService as Type<CheckoutStepService>
+      );
+    })
+  );
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(DeliveryModeComponent);
+    component = fixture.componentInstance;
+  });
+
+  it('should be created', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should get supported delivery modes', () => {
+    spyOn(
+      checkoutDeliveryModesFacade,
+      'getSupportedDeliveryModesState'
+    ).and.returnValue(
+      of({ loading: false, error: false, data: mockSupportedDeliveryModes })
+    );
+    component.ngOnInit();
+
+    component.supportedDeliveryModes$.subscribe((modes) => {
+      expect(modes).toBe(mockSupportedDeliveryModes);
+    });
+  });
+
+  it('should pre-select preferred delivery mode if not chosen before', () => {
+    spyOn(
+      checkoutDeliveryModesFacade,
+      'getSupportedDeliveryModesState'
+    ).and.returnValue(
+      of({ loading: false, error: false, data: mockSupportedDeliveryModes })
+    );
+    spyOn(
+      checkoutDeliveryModesFacade,
+      'getSelectedDeliveryModeState'
+    ).and.returnValue(of({ loading: false, error: false, data: undefined }));
+    spyOn(checkoutConfigService, 'getPreferredDeliveryMode').and.returnValue(
+      mockDeliveryMode1.code
+    );
+
+    component.ngOnInit();
+
+    expect(checkoutConfigService.getPreferredDeliveryMode).toHaveBeenCalledWith(
+      mockSupportedDeliveryModes
+    );
+    expect(component.mode.controls['deliveryModeId'].value).toBe(
+      mockDeliveryMode1.code
+    );
+  });
+
+  it('should select the delivery mode, which has been chosen before', () => {
+    spyOn(
+      checkoutDeliveryModesFacade,
+      'getSupportedDeliveryModesState'
+    ).and.returnValue(
+      of({ loading: false, error: false, data: mockSupportedDeliveryModes })
+    );
+    spyOn(
+      checkoutDeliveryModesFacade,
+      'getSelectedDeliveryModeState'
+    ).and.returnValue(
+      of({ loading: false, error: false, data: mockDeliveryMode2 })
+    );
+    spyOn(checkoutConfigService, 'getPreferredDeliveryMode').and.returnValue(
+      mockDeliveryMode1.code
+    );
+
+    component.ngOnInit();
+
+    expect(
+      checkoutConfigService.getPreferredDeliveryMode
+    ).not.toHaveBeenCalled();
+    expect(component.mode.controls['deliveryModeId'].value).toBe(
+      mockDeliveryMode2.code
+    );
+  });
+
+  it('should change step after invoking back()', () => {
+    component.back();
+    expect(checkoutStepService.back).toHaveBeenCalledWith(
+      <any>mockActivatedRoute
+    );
+  });
+
+  it('should get deliveryModeInvalid()', () => {
+    const invalid = component.deliveryModeInvalid;
+
+    expect(invalid).toBe(true);
+  });
+
```

Comment:
Leaving this for me to remember to check if the service for delivery mode has the assertion for testing for error, specifically JALO as retry error handling was removed from component level and moved to service level

```
  it('should reload delivery modes on error', () => {
    spyOn(
      mockCheckoutDeliveryService,
      'getLoadSupportedDeliveryModeProcess'
    ).and.returnValue(of({ loading: false, success: false, error: true }));
    component.ngOnInit();
    expect(
      mockCheckoutDeliveryService.loadSupportedDeliveryModes
    ).toHaveBeenCalledTimes(1);
  });
```